### PR TITLE
feat: add public v1 bootstrap endpoint

### LINKED ISSUE #1115: PR-2: Public v1 mobile bootstrap endpoint and instance validation
## 📌 Parent Epic

#1114

## 🎯 Goal

Implement the public `v1` bootstrap endpoint for the current deployment so Android can validate and bind to a customer-hosted SecPal instance before login.

## 📋 Scope Slice

This PR-sized slice is limited to serving the documented bootstrap contract from the API.

Included:
- public unauthenticated bootstrap route
- deployment-derived canonical runtime metadata
- failure responses for invalid or incompatible bootstrap state
- focused backend tests for success and failure paths

Explicitly out of scope:
- Android UI changes
- persistent app-side runtime binding
- tenant switching/reset logic
- customer-owned FCM runtime initialization

## 📋 Implementation Tasks

- [ ] Implement the public bootstrap endpoint defined by SecPal/contracts#281
- [ ] Source canonical runtime metadata from deployment configuration rather than hard-coded mobile defaults
- [ ] Return the canonical API base URL and deployment metadata needed before login
- [ ] Implement documented failure responses for incomplete bootstrap configuration, incompatible client/runtime version, or unsupported deployment state
- [ ] Fail closed when required bootstrap configuration is missing instead of silently falling back to legacy mobile assumptions
- [ ] Add Pest feature coverage for the successful bootstrap response and each documented failure slice
- [ ] Update backend/operator documentation for exposing bootstrap from customer-hosted deployments

## ✅ Acceptance Criteria

- [ ] An unauthenticated `v1` bootstrap request returns the documented runtime payload for a correctly configured deployment
- [ ] The response contains the canonical API base URL Android must use after bootstrap
- [ ] Missing or invalid bootstrap configuration produces a documented failure response rather than a hidden fallback to a baked-in production origin
- [ ] No secrets or server-only credentials are returned
- [ ] Pest coverage exists for the success path and the documented failure slices
- [ ] Breaking changes are allowed under the current `0.x` policy; legacy mobile bootstrap assumptions may be removed directly instead of being preserved behind compatibility shims

## 🔗 Dependencies

- Depends on: SecPal/contracts#281
- Blocks: SecPal/android#229, SecPal/android#230

## 📝 Technical Notes

- Keep the endpoint public and pre-login safe
- Prefer one canonical bootstrap route and serializer over multiple transitional variants
- Treat this as the server-side source of truth for runtime mobile binding

## 🧪 Testing Strategy

- Pest feature tests for successful bootstrap
- Pest feature tests for incompatible or invalid bootstrap states
- Focused assertions around canonical API URL derivation from deployment configuration

### PR DESCRIPTION
**Pull request by aroviqen**
## Summary
- add the public `GET /v1/bootstrap` endpoint with deployment-derived runtime metadata and documented `426` / `500` / `503` responses
- add bootstrap config, request validation, a dedicated throttle, and focused Pest coverage for success and failure paths
- document the required operator bootstrap environment variables and deployment verification flow for customer-hosted instances

## Testing
- php artisan test tests/Feature/Api/V1/BootstrapApiTest.php
- php artisan test tests/Feature/Api/V1/AndroidEnrollmentSessionApiTest.php --filter=bootstrap
- vendor/bin/pint --dirty
- vendor/bin/phpstan analyse app/Http/Controllers/Api/V1/BootstrapController.php --no-progress

## Scope
- keeps the change limited to the API-side public bootstrap contract
- leaves Android UI/runtime binding, tenant switching/reset, and customer-owned FCM initialization out of scope

## Links
- Closes #1115
- Part of #1114

---

> [!NOTE]
> **Medium Risk**
> Adds a new unauthenticated public API surface (`GET /v1/bootstrap`) with version gating and deployment-derived metadata, so misconfiguration or parsing edge cases could affect pre-login client behavior. Scoped rate limiting and focused tests reduce but don’t eliminate risk around exposing a new public endpoint.
> 
> **Overview**
> Adds a public `GET /v1/bootstrap` endpoint for Android pre-login runtime discovery, returning deployment-derived `api_base_url` (from `APP_URL`), instance display name, minimum supported app version/build, and pre-login feature flags.
> 
> Introduces `config/bootstrap.php` + documented `BOOTSTRAP_*` env vars, fail-closed error responses (`503` when disabled, `500` when required metadata is missing, `426` when the client is too old), and a dedicated `throttle:bootstrap` limiter. Includes Pest coverage for success, validation failures, rate limiting behavior, and the various fail-closed responses, plus deployment docs updates.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 40328a23529ea9861a413f5d6eeb9751439ca7ea. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: tests/Feature/Api/V1/BootstrapApiTest.php [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+use Illuminate\Foundation\Testing\RefreshDatabase;
+
+use function Pest\Laravel\getJson;
+
+uses(RefreshDatabase::class);
+
+beforeEach(function (): void {
+    config([
+        'app.name' => 'SecPal Demo',
+        'app.url' => 'https://customer-api.example/',
```
[the remark this answers is not in the document]
Comment (aroviqen) [reply tests/Feature/Api/V1/BootstrapApiTest.php:17]: Kept the external `customer-api.example` host intentionally here. The bootstrap contract is deployment-local runtime discovery, so a customer-controlled example origin makes it clearer that this endpoint is not tied to a SecPal-hosted domain.

File: docs/deployment.md [hunk 2]
```diff
@@ -132,6 +143,22 @@ LOG_LEVEL=warning
...
+- `APP_URL` must be the externally reachable API origin for this deployment. The endpoint derives the canonical `api_base_url` from `APP_URL` and appends `/v1`.
+- `APP_NAME` is used as the public instance display name unless `BOOTSTRAP_INSTANCE_DISPLAY_NAME` overrides it.
+- `BOOTSTRAP_MINIMUM_SUPPORTED_APP_VERSION` and `BOOTSTRAP_MINIMUM_SUPPORTED_APP_BUILD` are required. If either is missing, the endpoint fails closed with `500 BOOTSTRAP_STATE_INVALID` instead of falling back to SecPal-hosted defaults.
+- Set `BOOTSTRAP_PUBLIC_ENABLED=false` when the deployment should return the documented `503 BOOTSTRAP_CONFIG_UNAVAILABLE` response instead of serving bootstrap metadata.
+- `BOOTSTRAP_PASSWORD_LOGIN_ENABLED`, `BOOTSTRAP_PASSKEY_LOGIN_ENABLED`, and `BOOTSTRAP_MANAGED_ANDROID_ENROLLMENT_ENABLED` define the pre-login feature flags Android receives before login.
+
+Verify the deployed response with the canonical public origin:
+
+```bash
+curl --fail 'https://api.example.com/v1/bootstrap?client_platform=android&app_version=1.4.0&app_build=10400'
```
[the remark this answers is not in the document]
Comment (aroviqen) [reply docs/deployment.md]: Aligned the deployment guide on the same intentional customer-domain example (`customer-api.example`) so the operator docs stay consistent while still showing that bootstrap metadata is deployment-local rather than SecPal-hosted.

File: tests/Feature/Api/V1/BootstrapApiTest.php [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+});
+
+test('public bootstrap returns deployment-derived runtime metadata for a supported android client', function (): void {
+    getJson('/v1/bootstrap?client_platform=android&app_version=1.4.0&app_build=10400')
+        ->assertOk()
+        ->assertExactJson([
+            'data' => [
+                'client_platform' => 'android',
+                'api_base_url' => 'https://customer-api.example/v1',
+                'instance' => [
```
[the remark this answers is not in the document]
Comment (aroviqen) [reply tests/Feature/Api/V1/BootstrapApiTest.php:36]: Kept the external `customer-api.example` host intentionally here as well. For the bootstrap endpoint, a customer-domain example better signals that clients discover a deployment-specific API origin, not a shared SecPal service.
